src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Types;
+
+namespace Bicep.Core.TypeSystem.Providers.ThirdParty
+{
+    public class ThirdPartyResourceTypeProvider : ResourceTypeProviderBase, IResourceTypeProvider
```

Comment:
This class has many gaps in functionality mainly because we don't entirely have full clarity in what we need. This is more a flimsy barebones interpretation which needs to be built off of.